File: server/tests/fixtures/alpha_rule_triggers/typescript/architecture_dependency_cycle_impact.ts
// architecture.dependency_cycle_impact fixture
// This file is part of an impactful dependency cycle
// Should trigger: architecture.dependency_cycle_impact

import { cycleFunctionB } from './arch_dep_cycle_b';

export class CycleClassA {
    callB() {
        return cycleFunctionB();
    }
}

export function cycleFunctionA() {
    return new CycleClassA();
}

File: server/tests/fixtures/alpha_rule_triggers/python/architecture_dependency_cycle_impact.py
# architecture.dependency_cycle_impact fixture
# This file is part of an impactful dependency cycle
# Should trigger: architecture.dependency_cycle_impact

from arch_dep_cycle_b import cycle_function_b

class CycleClassA:
    """Part of a cycle that affects architecture."""
    
    def call_b(self):
        return cycle_function_b()
    
def cycle_function_a():
    """Function in the cycle."""
    return CycleClassA()
